### DISCUSSION TITLE
How to get client's IP address to block them from accessing API's?

### DISCUSSION THREAD

**Discussion by MariuzM** [vercel/next.js#20747] [Help]
I have tried this
```js
const Login = (props: any) => {
  return <div>Welcome to next.js! Your IP address is: {props.ip}</div>
}

export default Login

Login.getInitialProps = async ({ req, res }: any) => {
  const ip = req.connection.remoteAddress
  return { ip }
}
```
And it's deployed to Vercel but IP I'm getting is `127.0.0.1` which is coming from the server but not the user who opened that page. So I'm not sure if this is possible with the way Next.js works?

**Comment by dasblitz** [comment]
@MariuzM I know your question is old, but I came here through Google search for the same issue. I got it working by following the solution here: ~https://github.com/vercel/now-cloud/issues/60#issuecomment-470628187~

Edit: Since the link above is no longer available, I used this:

```
const ipAddress = req.headers["x-real-ip"]
```

**Reply by MacGyver98** [reply]
Oh man, the page is dead  May be could give as some instructions...

**Reply by jaintarun268** [reply]
@MariuzM github url is not working
